rclpy/rclpy/__init__.py

```diff
@@ -12,13 +12,37 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import rclpy._rclpy__rmw_opensplice_cpp as _rclpy
+import logging
+import os
+import sys
 
+from rclpy.impl import implementation_singleton
+from rclpy.impl import rmw_implementation_tools
 from rclpy.node import Node
 
+from rclpy.impl.implementation_singleton import rclpy_implementation as _rclpy
 
-def init(args):
-    return _rclpy.rclpy_init(args)
+
+def init(args=None):
+    rclpy_rmw_env = os.getenv('RCLPY_IMPLEMENTATION', None)
+    if rclpy_rmw_env is not None:
+        available_rmw_implementations = rmw_implementation_tools.get_rmw_implementations()
+        if rclpy_rmw_env not in available_rmw_implementations:
+            logger = logging.getLogger('rclpy')
+            logger.error(
+                "The rmw implementation specified in 'RCLPY_IMPLEMENTATION={0}', "
+                "is not one of the available implementations: {1}"
+                .format(rclpy_rmw_env, available_rmw_implementations)
+            )
+            sys.exit(1)
```

Comment:
Should this raise instead to enable calling this function, catch the exception and calling it again (e.g. with a different environment)?